### PR TITLE
Allow long running hidden

### PR DESCRIPTION
**Pull request by DeanArbel**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/23764

## Description
Added an allowed hidden parameter: *longRunning* to the hidden integration parameters validation.

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by DeanArbel** [comment]
@gal-berger fyi

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 1]
```diff
@@ -601,18 +601,21 @@ def is_valid_display_name(self):
...
 
-    def is_all_params_not_hidden(self) -> bool:
+    def is_valid_hidden_params(self) -> bool:
         """
         Verify there are no hidden integration parameters.
         Returns:
             bool. True if there aren't hidden parameters False otherwise.
         """
         ans = True
+        allowed_hidden_params = {'longRunning'}
```
Comment (Itay4) [inline demisto_sdk/commands/common/hook_validations/integration.py:611]: make this a const

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 2]
```diff
@@ -601,18 +601,21 @@ def is_valid_display_name(self):
...
         conf = self.current_file.get('configuration', [])
         for int_parameter in conf:
-            if int_parameter.get('hidden'):
+            hidden = int_parameter.get('hidden')
+            name = int_parameter.get('name')
```
Comment (Itay4) [inline demisto_sdk/commands/common/hook_validations/integration.py:615]:
```suggestion
            param_name = int_parameter.get('name')
```

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 3]
```diff
@@ -601,18 +601,21 @@ def is_valid_display_name(self):
                 return False
             return True
 
-    def is_all_params_not_hidden(self) -> bool:
+    def is_valid_hidden_params(self) -> bool:
         """
         Verify there are no hidden integration parameters.
```
Comment (Itay4) [inline demisto_sdk/commands/common/hook_validations/integration.py:606]: update docstring

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 4]
```diff
@@ -601,18 +601,21 @@ def is_valid_display_name(self):
...
         Returns:
             bool. True if there aren't hidden parameters False otherwise.
         """
         ans = True
+        allowed_hidden_params = {'longRunning'}
         conf = self.current_file.get('configuration', [])
         for int_parameter in conf:
-            if int_parameter.get('hidden'):
+            hidden = int_parameter.get('hidden')
```
Comment (Itay4) [inline demisto_sdk/commands/common/hook_validations/integration.py:614]:
```suggestion
            is_param_hidden = int_parameter.get('hidden')
```

File: demisto_sdk/commands/validate/tests/validators_test.py [hunk 5]
```diff
@@ -275,17 +275,17 @@ def test_is_root_connected_to_all_tasks(self, source, answer):
         finally:
             os.remove(PLAYBOOK_TARGET)
 
-    IS_ALL_PARAMS_NOT_HIDDEN_INPUTS = [
+    IS_VALID_HIDDEN_PARAMS = [
         (VALID_NO_HIDDEN_PARAMS, True),
```
Comment (Itay4) [inline demisto_sdk/commands/validate/tests/validators_test.py:279]: i dont understand how this is valid, it has hidden params, which are not `longRunning`

Comment (DeanArbel) [reply demisto_sdk/commands/validate/tests/validators_test.py:279]: it doesn't, those are `hidden: false` params

**Comment by Itay4** [comment]
lets add an integration test for this in https://github.com/demisto/demisto-sdk/blob/master/demisto_sdk/tests/integration_tests/validate_integration_test.py

File: demisto_sdk/tests/integration_tests/validate_integration_test.py [hunk 6]
```diff
@@ -5,50 +5,103 @@
...
+        - Ensure success validation message is printed.
+        """
+        pack_incident_field_path = join(AZURE_FEED_PACK_PATH, "IncidentFields/incidentfield-city.json")
+        runner = CliRunner(mix_stderr=False)
+        result = runner.invoke(main, [VALIDATE_CMD, "-p", pack_incident_field_path])
+        assert_positive(pack_incident_field_path, result)
+
+
+class TestIntegration:
+    @staticmethod
```
Comment (Itay4) [inline demisto_sdk/tests/integration_tests/validate_integration_test.py:45]: no need for the staticmethod decorators as these methods arent really called from anywhere

File: demisto_sdk/tests/integration_tests/validate_integration_test.py [hunk 7]
```diff
@@ -5,50 +5,103 @@
...
+        assert "The docker image tag is not the latest, please update it" in result.stdout
+        assert f"{pack_integration_path}: You're not using latest docker for the file, " \
+               "please update to latest version." in result.stdout
+        assert result.stderr == ""
+
+    @staticmethod
+    def test_negative__hidden_param():
+        """
+        Given
+        - Integration with not allowed hidden params.
```
Comment (Itay4) [inline demisto_sdk/tests/integration_tests/validate_integration_test.py:73]: please mention which at the hidden params that are not allowed

File: demisto_sdk/tests/integration_tests/validate_integration_test.py [hunk 8]
```diff
@@ -5,50 +5,103 @@
...
+
+        When
+        - Running validation on it.
+        Then
```
Comment (Itay4) [inline demisto_sdk/tests/integration_tests/validate_integration_test.py:77]: add white line here

File: demisto_sdk/tests/integration_tests/validate_integration_test.py [hunk 9]
```diff
@@ -5,50 +5,103 @@
...
+
+    @staticmethod
+    def test_positive__hidden_param():
+        """
+        Given
+        - Integration with allowed hidden params.
+
+        When
+        - Running validation on it.
+        Then
```
Comment (Itay4) [inline demisto_sdk/tests/integration_tests/validate_integration_test.py:98]: add white line here

File: demisto_sdk/tests/integration_tests/validate_integration_test.py [hunk 10]
```diff
@@ -5,50 +5,103 @@
...
+        - Ensure validation succeeds.
+        """
+        integration_path = join(TEST_FILES_PATH, 'integration-valid-no-unallowed-hidden-params.yml')
+        runner = CliRunner(mix_stderr=False)
+        result = runner.invoke(main, [VALIDATE_CMD, "-p", integration_path])
+        assert "Starting validating files structure" in result.stdout
+        assert f"Validating {integration_path}" in result.stdout
+        assert "can't be hidden. Please remove this field" not in result.stdout
```
Comment (Itay4) [inline demisto_sdk/tests/integration_tests/validate_integration_test.py:106]: why is this here?

Comment (DeanArbel) [reply demisto_sdk/tests/integration_tests/validate_integration_test.py:106]: want to validate it didn't encounter the error for hidden params while validating this file

File: demisto_sdk/tests/integration_tests/validate_integration_test.py (hunk 6 above)
Comment (DeanArbel) [reply demisto_sdk/tests/integration_tests/validate_integration_test.py:45]: true, it was mainly to remove the need for self. I'll remove those

**Comment by DeanArbel** [comment]
> i think that the integration tests you added will fail sometime when the docker image of it is gonna be updated

@Itay4 I think so too, that's why in the new tests I don't check if the validation is successful, rather that it didn't fail with the specific error I'm looking for.
I can adjust the old positive test to do the same.
